src/components/forms/LabForm/hooks/useLabForm.ts

```diff
@@ -1,32 +1,178 @@
+import { useLaboratoryProvider } from '@/providers/LaboratoryProvider/useLaboratoryProvider';
+import { Test } from '@/types/Test';
+import { SelectItem } from 'primereact/selectitem';
 import { useEffect, useState } from 'react';
+import { TestCategory } from '@/types/Test';
 
 interface LabFormData {
-  name?: string;
-  description?: string;
-  deadline?: Date;
+  name: string;
+  description: string;
+  deadline: string;
+  tests: Test[];
+  id?: number;
+  isHidden: boolean;
+  semesterNumber: number;
+
+  isEditing: boolean;
 }
 
-export const useLabForm = () => {
-  const [formData, setFormData] = useState<LabFormData>({});
+export const useLabForm = ({ isEditing, ...initialData }: LabFormData) => {
+  const [formData, setFormData] =
+    useState<Omit<LabFormData, 'isEditing'>>(initialData);
```

Comment:
isEditing должен быть статичным пропсом, который не меняется в рантайме. Не забывай, что при изменении любого стейта, реакт выполняет ререндер всего компонент и всех дочерних элементов. 

```suggestion
export const useLabForm = (isEditing: boolean, laboratory: Laboratory) => {
  const [formData, setFormData] = useState(laboratory);
```